spec/adapter_spec.rb

```diff
@@ -1,14 +1,55 @@
 require 'spec_helper'
 
-describe "adapter is set" do
-  it "should call it's enabled! method on init" do
+describe 'adapter is set' do
+
+  it 'should call the enabled! method on init' do
     QueueBus.send(:reset)
     adapter_under_test_class.any_instance.should_receive(:enabled!)
-    instance = adapter_under_test_class.new
+    adapter_under_test_class.new
     QueueBus.send(:reset)
   end
 
-  it "should be defaulting to Data from spec_helper" do
-    QueueBus.adapter.is_a?(adapter_under_test_class).should == true
+  it 'defaults to Data from spec_helper' do
+    QueueBus.adapter.is_a?(adapter_under_test_class).should eq(true)
+  end
+
+  context 'with no custom QueueBus Redis configuration' do
+
+    let(:resque_url) { 'redis://localhost:6379/0' }
+    let(:active_url) { ::Resque.redis.redis.client.options[:url] }
+
+    before(:each) do
+      Resque.redis = resque_url
+    end
+
+    it 'uses the default redis instance' do
+      QueueBus.adapter.redis do
+        active_url.should eq(resque_url)
+      end
+    end
+
+  end
+
+  context 'with a custom QueueBus Redis configuration' do
+    let(:resque_url) { 'redis://localhost:6379/0' }
+    let(:queuebus_url) { 'redis://localhost:6379/1' }
+    let(:active_url) { ::Resque.redis.redis.client.options[:url] }
```

Comment:
it might be a little tricky to use this in a `let` because it will cache the value for the second time even if it's not really true anymore.
It might be better to just check these explicitly in the tests instead of using a `let`.